library/core/src/convert/mod.rs

```diff
@@ -396,6 +396,7 @@ pub trait AsMut<T: ?Sized> {
 /// For example, take this code:
 ///
 /// ```
+/// # #![cfg_attr(not(bootstrap), allow(non_local_definitions))]
 /// struct Wrapper<T>(Vec<T>);
 /// impl<T> From<Wrapper<T>> for Vec<T> {
 ///     fn from(w: Wrapper<T>) -> Vec<T> {
```

Comment:
Adding the allow attribute seems to be necessary given the way rustdoc set up doc-tests.

Should we automatically add the attribute?

Reply:
Might be worth opening an issue in any case so it's not forgotten.

Reply:
Well I'm not sure if it's a bug or not. Does `rustdoc` put everything under a function? And are all doc-tests put under the same compilation unit?

Because if they are under the same compilation unit, it will affect other doc-tests, so we can't dismiss the lint and even if it's not there is still function in function.